gqlalchemy/models.py

```diff
@@ -111,8 +111,8 @@ def __init__(
         self,
         rel_id: Any,
         rel_type: str,
-        start_node: Node,
-        end_node: Node,
+        start_node: int,
```

Comment:
Why should nodes be expressed as integers?

Reply:
Because they are integers, when returning edges you get start_node and end_node as memgraph's internal ids

Reply:
It would make sense to implement lazy fetching when the user refers to the start/end node we would fetch it from memgraph (pro: you actually get what you want, con: getters could break). What do yous say?

Reply:
Do you think that it would lead toward a high number of database interactions? I imagine multiple users instantly iterating over edges and fetching nodes (because edge information itself is useless without starting and ending nodes).

Reply:
Yes it definitely would be an overhead, but there are two ways to solve this (as far as I know):
- We can load start/end node lazily when there are being fetched
- When executing a query that fetches edge, change it to fetch nodes as well and save them in edge data
I would leave it as is for now. since the user gets the start and end node in the same row, and can easily say which are start and end nodes.